dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/deaccession/DatasetDeaccessionServiceIT.java

```diff
@@ -0,0 +1,143 @@
+import javax.inject.Inject;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class DatasetDeaccessionServiceIT extends WebappArquillianDeployment {
+
+    @EJB
+    private DatasetDeaccessionService deaccessionService;
+
+    @Inject
+    private DatasetServiceBean datasetService;
+
+    @Inject
+    private DataverseSession dataverseSession;
+
+    @EJB
+    private AuthenticationServiceBean authenticationServiceBean;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void shouldDeaccessVersion() {
+        // given
+        Dataset dataset = datasetService.find(56L);
+        // when
+        DatasetVersion deaccessedVersion = deaccessionService.deaccessVersion(dataset.getReleasedVersion(), "TestReason", "https://www.google.com/");
+
+        // then
+        Dataset dbDataset = datasetService.find(56L);
+        assertEquals(deaccessedVersion.getVersionState(), DatasetVersion.VersionState.DEACCESSIONED);
+        assertTrue(dbDataset.getVersions().contains(deaccessedVersion));
+
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DEACCESSIONED));
+        assertFalse(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.RELEASED));
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.ARCHIVED));
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DRAFT));
+    }
+
+    @Test
+    public void shouldDeaccessVersions() {
+        // given
+        Dataset dataset = datasetService.find(56L);
+
+        // when
+        deaccessionService.deaccessVersions(dataset.getVersions(), "TestReason", "https://www.google.com/");
+
+        // then
+        Dataset dbDataset = datasetService.find(56L);
+        for (DatasetVersion version: dbDataset.getVersions()) {
+            assertEquals(DatasetVersion.VersionState.DEACCESSIONED, version.getVersionState());
+        }
+
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DEACCESSIONED));
```

Comment:
Shouldn't all of them be DEACCESSIONED?

Reply:
As above, it could be done in one assert and filtering that it contains only one VersionState.